lib/models/project.js

```diff
@@ -597,29 +598,37 @@ Project.closest = function(pathName, _ui, _cli) {
 Project.closestSync = function(pathName, _ui, _cli) {
   var ui = ensureUI(_ui);
   var directory, pkg;
-
   if (_cli && _cli.testing) {
     directory = existsSync(path.join(pathName, 'package.json')) && process.cwd();
     if (!directory) {
       if (pathName.indexOf(path.sep + 'app') > -1) {
-        directory = findupPath(pathName);
+        directory = findup.sync(pathName, 'package.json');
       } else {
         pkg = {name: 'ember-cli'};
       }
     }
-  } else {
-    directory = findupPath(pathName);
   }
-  if (!pkg) {
-    pkg = JSON.parse(fs.readFileSync(path.join(directory, 'package.json')));
+  else {
+    directory = findupPath(pathName, function(directory) {
+      var packageJsonPath = path.join(directory, 'package.json');
+      var isPackageJsonPresent = existsSync(packageJsonPath);
+      if (!isPackageJsonPresent) {
+        return false;
+      }
+      var localPkg = require(packageJsonPath);
```

Comment:
lets readFileSync + json.parse or we will cache these files, which i don't think we want to.